public/js/components/UIMenu.js

```diff
@@ -81,73 +81,108 @@ class UIMenu extends UIComponent {
     let activeMenuItems
     let activeMenuItemsSel
     let button
-    let dataSource
-    let parentNode
+    let item
 
     for (; i<l; i++) {
       buttons[i].addEventListener('click', evt => {
         evt.preventDefault()
 
         button = evt.target
-        dataSource = button.getAttribute('data-source')
-        parentNode = button.parentNode
-
-        activeMenuItemsSel = `#${this.domNode.id} .menu__item--active:not(#${parentNode.id})`
+        item = button.parentNode
+        activeMenuItemsSel = `#${this.domNode.id} .menu__item--active:not(#${item.id})`
         activeMenuItems = document.querySelectorAll(activeMenuItemsSel)
 
         if (activeMenuItems.length) {
-          this.hide(activeMenuItems)
+          this.hideItems(activeMenuItems)
         }
 
-        this.show(parentNode, dataSource)
+        this.showItem(item)
+        // @TODO manage widget state based on History API
+        //this.updateWidgetState(item)
       })
     }
   }
 
-  show(item, dataSource) {
-    let visible = item.classList.contains('menu__item--active')
-    let toggle = visible ?  'remove' : 'add'
-    let detail = item.children[0].nextSibling
+  hideItems(items) {
+    items.forEach(item => {
+      item.classList.remove('menu__item--active')
+      this.setDetailVisibility(item.children[1])
+    })
+  }
 
-    if (!visible) {
-      let cachedItem = sessionStorage.getItem(dataSource)
-      let itemIndex = Array.from(item.parentNode.children).indexOf(item)
-      let title = `${document.title} - `
+  showItem(item) {
+    const active = this.getItemState(item)
+    let detail = item.children[1]
 
-      if (!cachedItem) {
-        api.get(dataSource).then(data => {
-          this.buildItemDetail(data, item)
-          this.store(dataSource, JSON.stringify(data))
-          item.children[1].setAttribute('aria-hidden', visible)
-          title += data.name
-        })
-      } else {
-        let cachedData = JSON.parse(cachedItem)
+    this.setItemState(item)
 
+    if (active) {
+      this.setDetailVisibility(detail)
+    } else {
+      this.checkItemCache(item).then(data => {
```

Comment:
As *checkItemCache* promise may be rejected if an error is catched after API request (line 150), error handling should be subsequently handled here.